src/homework/db/crud.py

```diff
@@ -98,3 +98,85 @@ def get_user_balance_by_id(user_id: int) -> None | int:
             return balance
         except (exc.SQLAlchemyError, exc.DBAPIError):
             return None
+
+
+def calculate_new_points(
+    total_cost: int, num_of_deleted_orders: int, total_num_of_orders: int
+) -> int:
+    """
+    Calculate the number of additional points.
+
+    :param total_cost: int
+    :param num_of_deleted_orders: int
+    :param total_num_of_orders: int
+    :return: int
+    """
+    normal_coef = min(
+        1, 1.2 - num_of_deleted_orders / (total_num_of_orders + 1)
+    )
+    new_points = total_cost * 0.05 * normal_coef
+    return int(new_points)
+
+
+def create_order_update_balance(
```

Comment:
In general, bounding two different SQL operation in one function, might not be a good idea. However, due to the documentation and the project's logic this method is fine, as order creation is always followed by the balance update. Although, if the fundamental logic of the project will change in the future, this method will be the first candidate for the refactoring. Overall, now it is okay and let's keep it, good job!

Reply:
Yes, I completely agree with you.